pymc3/tests/test_ode.py

```diff
@@ -0,0 +1,474 @@
+                                         n_odeparams=1,
+                                         n_states=1)
+
+        integrated_solution, *_ = ode_model._simulate([alpha, y0])
+
+        manual_logp = norm.logpdf(x=np.ravel(yobs), loc=np.ravel(integrated_solution), scale=1).sum()
+
+        with pm.Model() as model_1:
+            forward = ode_model(odeparams=[alpha], y0=[y0]).reshape(yobs.shape)
+            y = pm.Normal('y', mu=forward, sd=1, observed=yobs)
+
+        pymc3_logp = model_1.logp()
+        np.testing.assert_allclose(manual_logp, pymc3_logp)
+
+class TestErrors(object):
+
+    '''Test running model for a scalar ODE with 1 parameter'''
+    def system(y, t, p):
+        return np.exp(-t) - p[0] * y[0]
+
+    times = np.arange(0, 9)
+
+    ode_model = DifferentialEquation(func=system,
+                                    t0=0,
+                                    times=times,
+                                    n_states=1,
+                                    n_odeparams=1)
+
+    def test_too_many_params(self):
+        with pytest.raises(ValueError):
+            self.ode_model(odeparams=[1, 1], y0=[0])
+
+    def test_too_many_y0(self):
+        with pytest.raises(ValueError):
+            self.ode_model(odeparams=[1], y0=[0, 0])
+
+    def test_too_few_params(self):
+        with pytest.raises(ValueError):
+            self.ode_model(odeparams=[], y0=[1])
+
+    def test_too_few_y0(self):
+        with pytest.raises(ValueError):
+            self.ode_model(odeparams=[1], y0=[])
+
+    def test_func_callable(self):
+        with pytest.raises(ValueError):
+            DifferentialEquation(func = 1,
+                                 t0 = 0, 
+                                 times = self.times,
+                                 n_states = 1,
+                                 n_odeparams = 1)
+
+    def test_number_of_states(self):
+        with pytest.raises(ValueError):
+            DifferentialEquation(func = self.system,
+                                 t0 = 0, 
+                                 times = self.times,
+                                 n_states = 0,
+                                 n_odeparams = 1)
+
+    def test_number_of_params(self):
+        with pytest.raises(ValueError):
+            DifferentialEquation(func = self.system,
+                                 t0 = 0, 
+                                 times = self.times,
+                                 n_states = 1,
+                                 n_odeparams = 0)
+
+class TestDiffEqModel(object):
+
+    def test_scalar_ode_1_param(self):
+        '''Test running model for a scalar ODE with 1 parameter'''
+        with theano.configparser.change_flags(compute_test_value='off'):
+            def system(y, t, p):
+                return np.exp(-t) - p[0] * y[0]
+
+            times = np.array([0.5, 1., 1.5, 2., 2.5, 3., 3.5,
+                              4., 4.5, 5., 5.5, 6., 6.5, 7., 7.5])
+
+            yobs = np.array([0.31,
```

Comment:
...is this `black`'s doing? i might make it one line otherwise

Reply:
I think that is VS code trying to be helpful.